lib/new_relic/agent/instrumentation/grpc/client/instrumentation.rb

```diff
@@ -0,0 +1,88 @@
+# encoding: utf-8
+# frozen_string_literal: true
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative 'request_wrapper'
+require_relative '../helper'
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      module GRPC
+        module Client
+          include NewRelic::Agent::Instrumentation::GRPC::Helper
+
+          def issue_request_with_tracing(grpc_type, method, requests, marshal, unmarshal,
+            deadline:, return_op:, parent:, credentials:, metadata:)
+            return yield unless trace_with_newrelic?
+
+            segment = request_segment(method)
+            request_wrapper = NewRelic::Agent::Instrumentation::GRPC::Client::RequestWrapper.new(@host)
+            segment.add_request_headers(request_wrapper)
+            metadata.merge!(metadata, request_wrapper.instance_variable_get(:@newrelic_metadata))
+            grpc_message = nil
+            grpc_status = 0
+
+            NewRelic::Agent.disable_all_tracing do
```

Comment:
Why is this method being called?

Reply:
Looking in the codebase, I see a similar use case for its use around something noticing an error to avoid double tracing. Is that the purpose here too?

Reply:
This pattern was copied from `net_http/instrumentation.rb`, which was used as a reference instrumentation implementation. My hunch was that the agent ought to be completely out of the way of the original instrumented method and not inadvertently interfere with it. But the prevention of duplication makes sense.